bp_be/src/include/bp_be_internal_if_defines.vh

```diff
@@ -189,7 +190,7 @@
    )
  
 `define bp_be_trap_pkt_width(vaddr_width_mp) \
-  (2 * vaddr_width_mp + rv64_priv_width_gp + 4)
+  (2 * vaddr_width_mp + rv64_priv_width_gp + dword_width_p + 4)
```

Comment:
Can we make this smaller than a full word?

Reply:
I don't think so, because the high bit indicates whether it's an interrupt.

Reply:
Actually, since we only notify Dromajo on interrupts(not sync exceptions), I can only send the low bits to the nonsynth_cosim  module and recreate the whole 64-bit over there. But what's the concern here? Because it's gonna be optimized out in any synthesis since it's not used, right?

Reply:
It's a good question.  @taylor-bsg and I had a discussion on this a ways back.  Basically, how much should we rely on synthesis tools? 

One school of thought is to forbid wires only intended for nonsynth use, to make the hardware intent explicit.

Another is that, like you say, those wires should be ripped out.  However, different synth settings could lead to extra buffer insertion, i.e. if this packet is on a feedthrough path, without flattening enabled.

The other question is if we ever think mcause will be used in the trap packet, for example to send different types of messages to the front end.

Reply:
I don't think it'll be used in the logic unless they change smth in the riscv spec.

Reply:
How about just using mcause_li?

Reply:
It's a little less clean in the cosim module, but I'd prefer it over adding unnecessary wires

Reply:
Actually, forget it.  This might actually make another change I'm planning easier